src/policy/policy.cpp

```diff
@@ -135,7 +135,7 @@ bool IsStandardTx(const CTransactionRef tx, std::string &reason)
     return true;
 }
 
-bool AreInputsStandard(const CTransactionRef tx, const CCoinsViewCache &mapInputs)
+bool AreInputsStandard(const CTransactionRef tx, const CCoinsViewCache &mapInputs, bool may2020Enabled)
 {
```

Comment:
This is a nit but still I decided to state it :smile: 

I found confusing the use of boolean flag passed as parameter here. Why not just using the helper in `src/validation/forks.{h,cpp}`?

Especially because sometimes we define the flag a "global" to a given unit, see miner.cpp, in some other situation like the one here we passing it via parameter to a function.

Reply:
Using the forks functions requires a CBlockIndex pointer.  I could use chainActive->tip(), relying on global state is considered bad form, especially the deeper you get into software layers.  The practical reason behind this rule is because doing so pins the entire implementation to a single chain.  If there was no chainActive (and coins) globals, but instead some object that you "new", you can see how it would be very easy to simultaneously fully track multiple tips -- you'd just instantiate multiple copies of that object and call the relevant member functions.

I could pass CBlockIndex rather then a boolean.  However, doing so means that the policy library now relies on the block index data structures.  I want to keep subsystems separate unless they must be joined due to extensive cross-use.

Reply:
fair enough, thanks for the explanation